src/lib/game/world/map.js

```diff
@@ -130,12 +130,17 @@ class WorldMap extends THREE.Group {
         entry.position.y
       );
 
+      // Provided as (Z, X, -Y)
       m2.rotation.set(
+        entry.rotation.z * Math.PI / 180,
         entry.rotation.x * Math.PI / 180,
-        -entry.rotation.z * Math.PI / 180,
-        entry.rotation.y * Math.PI / 180
+        -entry.rotation.y * Math.PI / 180
       );
 
+      // Adjust M2 rotation to match Wowser's axes.
+      const quat = m2.quaternion;
+      quat.set(quat.x, quat.y, quat.z, -quat.w);
+
```

Comment:
Ah yeah. No, I meant setting the quaternion directly, but I suppose it's the flipping of `w` that causes trouble, seeing as we don't know what to set it to?